cdap-api/src/main/java/co/cask/cdap/api/annotation/Requirements.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.api.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * <p>Annotates the requirements needed by a plugin to run successfully.</p>
+ *
+ * <p>Requirements are case insensitive.</p>
+ *
+ * <p>If a plugin does not specify any {@link Requirements} or the {@link #value()} array is empty then it assumed
+ * that the plugin does not have any specific requirements and can run everywhere.</p>
+ *
+ * <p>Usage Examples:</p>
+ * <ul>
+ * <li><b>Omitting Requirements:</b> A plugin can chose to not specify any requirement by not specifying a
+ * {@link Requirements} annotation. In this case the plugin will considered to be capable or run everywhere.</li>
+ * <pre>
+ *     {@literal @}Plugin(type = BatchSource.PLUGIN_TYPE)
+ *     {@literal @}Name("NoRequirementSource")
+ *      public class NoRequirementSource extends{@code BatchSource<byte[], Row, StructuredRecord>} {
+ *       ...
+ *       ...
+ *      }
+ *   </pre>
+ * <li><b>Specifying a particular requirement:</b> If a plugin is capable to run only when 'transactions' are
+ * available then this can be specified as below.</li>
+ * <pre>
+ *     {@literal @}Plugin(type = BatchSource.PLUGIN_TYPE)
+ *     {@literal @}Name("TransactionalSource")
+ *     {@literal @}Capability({"transactions"})
+ *      public class TransactionalSource extends{@code BatchSource<byte[], Row, StructuredRecord>} {
+ *       ...
+ *       ...
+ *      }
+ *   </pre>
+ * <li><b>Specifying multiple requirements:</b> A plugin can also specify multiple requirements. For example if a
+ * plugin needs 'transactions' and 'kms' to run this can specified as below.</li>
```

Comment:
lets use 'securestore' instead of 'kms' to be consistent with terminology.

Reply:
Done.